sdk/dotnet/Makefile

```diff
@@ -1,18 +1,42 @@
-PROJECT_NAME       := Pulumi .NET Core SDK
-LANGHOST_PKG       := github.com/pulumi/pulumi/sdk/dotnet/cmd/pulumi-language-dotnet
-VERSION            := $(shell ../../scripts/get-version)
-VERSION_DOTNET     := ${VERSION:v%=%}                             # strip v from the beginning
-VERSION_PREFIX     := $(firstword $(subst -, ,${VERSION_DOTNET})) # e.g. 1.5.0
-VERSION_SUFFIX     := $(word 2,$(subst -, ,${VERSION_DOTNET}))    # e.g. alpha.1
-PROJECT_PKGS       := $(shell go list ./cmd...)
-PULUMI_LOCAL_NUGET ?= ~/.nuget/local
+PROJECT_NAME         := Pulumi .NET Core SDK
+LANGHOST_PKG         := github.com/pulumi/pulumi/sdk/dotnet/cmd/pulumi-language-dotnet
+
+PROJECT_PKGS         := $(shell go list ./cmd...)
+PULUMI_LOCAL_NUGET   ?= ~/.nuget/local
+
+VERSION              := $(shell ../../scripts/get-version)
+VERSION_DOTNET       := ${VERSION:v%=%}                                   # strip v from the beginning
+VERSION_FIRST_WORD   := $(strip $(word 1,$(subst -, ,${VERSION_DOTNET}))) # e.g. 1.5.0
+VERSION_SECOND_WORD  := $(strip $(word 2,$(subst -, ,${VERSION_DOTNET}))) # e.g. alpha or alpha.1
+VERSION_THIRD_WORD   := $(strip $(word 3,$(subst -, ,${VERSION_DOTNET}))) # e.g. featbranch or featbranch.1
+
+VERSION_PREFIX       := ${VERSION_FIRST_WORD}
+
+ifeq ($(VERSION_THIRD_WORD),)
+	VERSION_SUFFIX   := ${VERSION_SECOND_WORD}
+else
+	# have to use addsuffix or make adds a whitespace between the two values.
+	VERSION_SUFFIX   := $(addsuffix -$(VERSION_THIRD_WORD),${VERSION_SECOND_WORD})
+endif
 
 TESTPARALLELISM := 10
 
 include ../../build/common.mk
 
 build::
-	dotnet build dotnet.sln /p:VersionPrefix=${VERSION_PREFIX} /p:VersionSuffix=${VERSION_SUFFIX}
+	# .NET support is only in Preview currently.  So unilaterally add `-preview` to the version suffix.
```

Comment:
Every version suffix, inluding `alpha` already means preview package. Moreover, as `preview` is alphabetically higher than `alpha` or `beta`, `preview*` packages will take precedence. I suggest we remove it.

Reply:
the problem is that when we move normal pulumi 1.5 out of alpha we dont' want to move .net out of alpha.

`preview` is hte highest order bit for the `dotnet sdk` so i think this is appropriate.